src/utilities/types.ts

```diff
@@ -60,3 +60,20 @@ type NormalizeKeyPathsRecursively<T, K = keyof T> = K extends keyof T
  * Normalize read-only key paths
  */
 export type NormalizeKeyPaths<T> = keyof T | NormalizeKeyPathsRecursively<T>;
+
+/**
+ * Compose a new type from another type by mapping snake case type keys to camel case
+ * E.g. type T = { key_of_string: string } => type U = { keyOfString: string }
+ *
+ * Inspired by:
+ * https://stackoverflow.com/a/65015868/3431443
+ */
+type CamelCase<S extends string> = S extends `${infer P1}_${infer P2}${infer P3}`
+  ? `${Lowercase<P1>}${Uppercase<P2>}${CamelCase<P3>}`
+  : Lowercase<S>;
+
+export type KeysToCamelCase<T> = {
+  [K in keyof T as CamelCase<string & K>]: T[K] extends Record<string, unknown>
+    ? KeysToCamelCase<T[K]>
+    : T[K];
+};
```

Comment:
looks good. any change we can test types?

Reply:
I'm currently working on a js core feature: getTokenByRefreshToken
We will see the actual usage in that PR by EOD today.

Reply:
Unit test cases will be added shortly in a follow-up PR.
[Linear issue link](https://linear.app/silverhand/issue/LOG-1428/add-unit-tests-for-log-1427)